crates/bevy_ecs/src/world/mod.rs

```diff
@@ -1274,12 +1274,30 @@ impl World {
         }
     }
 
+    /// Runs both [`clear_entities`](Self::clear_entities) and [`clear_entities`](Self::clear_resources),
+    /// invalidating all [Entity] and resource fetches such as [`Res`](crate::system::Res), [`ResMut`](crate::system::ResMut)
+    pub fn clear_all(&mut self) {
+        self.clear_entities();
+        self.clear_resources();
+    }
+
+    /// Clears entities in this [World] including entities in [Archetypes].
+    /// All [Entity] will be invalidated.
     pub fn clear_entities(&mut self) {
         self.storages.tables.clear();
         self.storages.sparse_sets.clear();
         self.archetypes.clear_entities();
         self.entities.clear();
     }
+
+    /// Clears all resources in this [World].
+    /// Any resource fetch to this [World] will fail unless they are re-initialized.
```

Comment:
```suggestion
    ///
    /// Any resource fetch to this [World] will fail unless they are re-initialized.
```

Line break between summary and details by convention.